ui/pages/confirmations/components/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js

```diff
@@ -59,14 +72,41 @@ function BlockaidBannerAlert({ txData, ...props }) {
 
   const t = useContext(I18nContext);
   const { updateTransactionEventFragment } = useTransactionEventFragment();
+  const [isLoadingDisplayed, setIsLoadingDisplayed] = useState(false);
+
+  useEffect(() => {
+    if (securityAlertResponse?.reason === 'loading') {
+      setIsLoadingDisplayed(true);
+    }
+  }, [securityAlertResponse]);
```

Comment:
```suggestion
  }, [securityAlertResponse.reason]);
```
suggestion to avoid unnecessary rerenders

Reply:
I do not think we change `securityAlertResponse`.

Reply:
agree not needed though I think it would be nice to have